### ISSUE TITLE
chrony does not want 0.opnsense.pool.ntp.org to be removed

### ISSUE THREAD

**Issue by erdnuesse** [opnsense/plugins#4326]
[X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.m
[X] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
[X] The title contains the plugin to which this issue belongs

**Describe the bug**
During net-wide chrony rollout I discovered the following: 
Trying to remove servers from the list seems to be buggy. 
As in: removing "0.opnsense.pool.ntp.org" and saving; despite having 6 working alternatives in the list throws: 
```
Input validation
Please correct validation errors in form
```

- no chance to validate regressions. 

**To Reproduce**
1. Go to Services > Chrony > General
2. Click on the X to remove the opnsense entry
3. Save
![image](https://github.com/user-attachments/assets/53afa1c6-e95d-443e-9ae0-94333a6ae138)
![image](https://github.com/user-attachments/assets/199dd7ed-a4a2-4c8d-a7c5-f2ac7837dbb0)

 **Expected behavior**
Regular save; without error, just no longer the opnsense entry.

**Relevant log files**
However, I do not see logs regarding the saving of the config, just entries that pushed me to remove the servers in the first place: 
```
2024-10-27T10:56:21	Error	chronyd	NTS-KE session with 17.253.14.251:4460 (0.opnsense.pool.ntp.org) timed out	
2024-10-27T10:49:04	Error	chronyd	Could not connect to 107.189.12.98:4460 (0.opnsense.pool.ntp.org) : Connection refused
```
No errors in the Web GUI Log either.

**Additional context**
I'm not saying the validation does not need to be fixed, but I wouldn't have tried if I hadn't seen the errors with the opnsense servers (which may need fixing? idk) in the first place, but that's another story.

However; commenting out 
~ # vi /usr/local/etc/chrony.conf
`#server 0.opnsense.pool.ntp.org iburst nts`
results in chronyd obviously omitting and no longer retrying the opnsense pool, but the web interface still gives me
![chronyd](https://github.com/user-attachments/assets/e6064b6a-938f-4296-a039-3715eb17b25e)
(I even deleted the line, same result)
And obviously restarted the service each time, also, same result. 
It seems to be the validation checking for a specific expected entry; and not accepting alternate servers exclusively. That's as far as I can narrow it down. 

**Environment**
chrony plugin 1.5
OPNsense 24.7.7 amd64; Commit: f20b6eaa5
AMD GX-412TC SOC (4 cores, 4 threads)

**Comment by erdnuesse** [comment]
What needs to be done to get a fix for this? As in: how do I find the underlying code, probably review and suggest changes in the codebase? 
I'm not a coder myself, but I am an IT professional, and reading the code and pseudo-coding the change should be possible.

**Comment by TheNec** [comment]
I don't know if that helps but I changed to Chrony some weeks ago.

According to OPNsense's changelogs there was no update to the Chrony-Plugin in the last months but that doesn't mean that there wasn't really one that may not have been mentioned and your screenshot looks a bit different from my OPNsense but I have browser darkmode activated.

The last change to 'master' was about 6 months ago: [OPNsense chrony source (underlying code)](https://github.com/opnsense/plugins/tree/master/net/chrony)

1. I wanted to use the Pool-NTP-Servers but I checked it and it seems that they don't support NTS. I don't know if the OPNsense NTPs do but most probably that's where the errors in the log come from. Btw. I think there may be a bug somewhere because while your Chrony tries to use NTS-Port 4460 for NTP-Servers(?) my Chrony uses NTP-Port 123 for NTS-Servers which work anyways in NTS-Mode (and it seems that the fallback peer is treated as any other server becoming the active peer very often maybe for using an IP-Address and not needing to DNS-Resolve before connecting)

2. Instead of using the 'x' on the entry, move your mousepointer to the right from the entries until the cursor appears and click into the field. Then use backspace to delete the entries or try the 'Clear all' button. You may also use the 'Text' button especially when adding entries (one per line!). I'm not sure but I think I remember also needing to try around there.

3. You may try to reboot the device after you manually editing the 'chrony.conf' file

4. If this doesn't help you may un-/re-/install the chrony plugin but I it's possible that you need to delete the configuration files manually after uninstalling before installing again.

HTH

**Comment by erdnuesse** [comment]
Hi, regarding your points: 
I don't say it's a regression. maybe it never worked properly (as expected for my use-case). Or no one except me ever tried to use their list own list of ntp servers only, without those from opnsense. 
To...
1: NTS - I have no Idea what you want to say with that, the problem is me trying to remove one of the servers in the ntp peers list. I frankly don't care about the capabilities of that specific peer. If you think that was the only reason for me to remove it - no, it wasn't and possibly deflects from my issue here.
2. Does it... / or better: should it matter as to whether I use backspace, delete, the [x] or ctrl-X to empty the field, in any case it is not a user error. If anything it's a bug, as the UI provides a way to configure something, but the error says it clearly does not allow for what I want to achieve. - Though I accept that would be good for narrowing down on the issue, but since I am not a Dev, I primarily focus on reporting bugs - not necessarily fixing them or working around them. So if anyone has a working feature (ie. having an empty list with only their own choice of tnp peers, without the opnsense one); I'd be happy to do so, my guess is, no one ever tried that, hence my guess this needs fixing. 
3. got it. 
4. It was a clean install - at least for the plugin - for me; so first usage of the plugin, therefore I expect to have already started with a clean config file. 

Thanks; 
However, read again the section I already stated: 

> However; commenting out
> ~ # vi /usr/local/etc/chrony.conf
> #server 0.opnsense.pool.ntp.org iburst nts
> results in chronyd obviously omitting and no longer retrying the opnsense pool, but the web interface still gives me
> chronyd
> (I even deleted the line, same result)
> And obviously restarted the service each time, also, same result.
> It seems to be the validation checking for a specific expected entry; and not accepting alternate servers exclusively. That's as far as I can narrow it down.

And if everyone needs a workaround, and bugs get closed without anyone addressing them, this project would be going downhill **_fast_**
So I conclude;
IMO, this isn't fixed at all, and since there was no reason given, I expect it to become as such.

The tl;dr is: 
why does opnsense do not want to fix the GUI even when I remove their NTP servers from my config? I can't find a single reason. 

My fix: Spinning up a chrony docker in my infra, and omitting the feature on the gateway entirely. I don't want to deal with broken features, so I rather set up my own - and don't want to deal with this anymore. I'm willing to report bugs and help with logs and everything, but if that's the way this project is managed, that's as far as I go. This one is closed, I use another service, I no longer need to care more than any maintainer does.

- K - out.
